protocol/messenger_test.go

```diff
@@ -936,6 +936,73 @@ func (s *MessengerSuite) TestChangeNameGroupChat() {
 	s.Require().Equal(newName, actualChat.Name)
 }
 
+// Test being re-invited to a group chat
+func (s *MessengerSuite) TestReInvitedToGroupChat() {
+	var response *MessengerResponse
+	theirMessenger := s.newMessenger(s.shh)
+	response, err := s.m.CreateGroupChatWithMembers(context.Background(), "old-name", []string{})
+	s.NoError(err)
+	s.Require().Len(response.Chats, 1)
+
+	ourChat := response.Chats[0]
+
+	err = s.m.SaveChat(ourChat)
+	s.NoError(err)
+
+	members := []string{"0x" + hex.EncodeToString(crypto.FromECDSAPub(&theirMessenger.identity.PublicKey))}
+	_, err = s.m.AddMembersToGroupChat(context.Background(), ourChat.ID, members)
+	s.NoError(err)
+
+	// Retrieve their messages so that the chat is created
+	err = tt.RetryWithBackOff(func() error {
+		var err error
+		response, err = theirMessenger.RetrieveAll()
+		if err == nil && len(response.Chats) == 0 {
+			err = errors.New("chat invitation not received")
+		}
+		return err
+	})
+	s.Require().NoError(err)
+
+	_, err = theirMessenger.ConfirmJoiningGroup(context.Background(), ourChat.ID)
+	s.NoError(err)
+
+	// Wait for join group event
+	err = tt.RetryWithBackOff(func() error {
+		var err error
+		response, err = s.m.RetrieveAll()
+		if err == nil && len(response.Chats) == 0 {
+			err = errors.New("no joining group event received")
```

Comment:
We use this `RetryWithBackOff()` block a lot in these tests, wouldn't it make sense to refactor it out as a helper for waiting with a custom error message as argument?